clients/src/test/java/org/apache/kafka/common/serialization/SerializationTest.java

```diff
@@ -119,6 +120,49 @@ public void testLongSerializer() {
     }
 
     @Test
+    public void testFloatSerializer() {
+        Float[] floats = new Float[]{
+            5678567.12312f,
+            -5678567.12341f
+        };
+
+        Serializer<Float> serializer = Serdes.Float().serializer();
+        Deserializer<Float> deserializer = Serdes.Float().deserializer();
+
+        for (Float value : floats) {
+            assertEquals("Should get the original double after serialization and deserialization",
+                    value, deserializer.deserialize(topic, serializer.serialize(topic, value)));
+        }
+
+        assertEquals("Should support null in serialization and deserialization",
+                null, deserializer.deserialize(topic, serializer.serialize(topic, null)));
+
+        serializer.close();
+        deserializer.close();
+    }
+
+    @Test(expected = SerializationException.class)
```

Comment:
We have a rule w.r.t using `@expected` the rule is that it should only be used for single line tests where you can guarantee that the exception can only come from the line that is being executed. In all other cases you should use `try{...}catch(...)`